test/api/capi/test_capi_extract.cpp

```diff
@@ -0,0 +1,42 @@
+#include "capi_tester.hpp"
+
+using namespace duckdb;
+using namespace std;
+
+TEST_CASE("Test extract statements in C API", "[capi]") {
+	CAPITester tester;
+	duckdb_result res;
+	duckdb_extracted_statements stmts = nullptr;
+	duckdb_state status;
+
+	REQUIRE(tester.OpenDatabase(nullptr));
+
+	idx_t size = duckdb_extract_statements(tester.connection,
```

Comment:
Could we add a test with a parser failure in the `extract_statements`?

What about if the query is the empty string?